amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShutdownTask.java

```
@@ -179,7 +175,28 @@ public TaskResult call() {
         }
 
         return new TaskResult(exception);
+    }
+
+    private void createLeasesForChildShardsIfNotExist()
+            throws RuntimeException, DependencyException, InvalidStateException, ProvisionedThroughputException {
```

Let's avoid throwing unchecked exception. You can throw a checked exception from createNewLeaseForShard

---

Somehow included the RuntimeException by accident. Will remove